dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/registry/ZoneAwareClusterInvoker.java

```diff
@@ -216,23 +209,16 @@ public Result doInvoke(Invocation invocation, final List<Invoker<T>> invokers, L
 
     private boolean shouldMigrate(boolean addressChanged, List<Invoker<T>>  serviceInvokers, List<Invoker<T>>  interfaceInvokers) {
         Set<MigrationClusterComparator> detectors = ExtensionLoader.getExtensionLoader(MigrationClusterComparator.class).getSupportedExtensionInstances();
-        if (null != detectors && detectors.size() > 0) {
-            if (detectors.stream().allMatch(s -> s.shouldMigrate(interfaceInvokers, serviceInvokers))) {
-                return  true;
-            } else {
-                return false;
-            }
-        } else {
-            List<Invoker<T>>  availableServiceInvokers = serviceInvokers.stream().filter( s -> ((MigrationClusterInvoker)s).isAvailable()).collect(Collectors.toList());
-            if (availableServiceInvokers.isEmpty()) {
-                return  false;
-            } else {
-                return  true;
-            }
+        if (detectors != null && !detectors.isEmpty()) {
+            return detectors.stream().allMatch(s -> s.shouldMigrate(interfaceInvokers, serviceInvokers));
         }
+
+        // check application level provider available.
+        List<Invoker<T>>  availableServiceInvokers = serviceInvokers.stream().filter( s -> s.isAvailable()).collect(Collectors.toList());
```

Comment:
indent can be format followed the checkstyle.

Reply:
fixed it.